include/zenoh-pico/link/link.h

```diff
@@ -105,6 +101,7 @@ typedef struct _z_link_t {
 
     uint16_t _mtu;
     uint8_t _capabilities;
+    bool _is_reliable;
```

Comment:
`_is_reliable` should be a bit in the `_capabilities`.

Reply:
_capabilities is no longer a bitfield though, eventually capabilities could be a structure including an enum and the reliable flag

Reply:
The idea of the bitfield was to reduce the overhead. These values will be booleans, so they can be store in a single bit.
If this was discussed within the team, you can ignore. Otherwise, I suggest to be rethought because this was done intentionally.

Reply:
tagging @p-avital

Reply:
Also invoking @Mallets

Reply:
In this specific case I don't think it will make any difference. 
`capabilities` is `uint8_t` and `bool` takes the same space of a `uint8_t`. 
Unless you are running on a 8-bit platform, and because of alignment, the two solutions are equivalent w.r.t. space usage. 

The bitfield usually is used to express the combination of different independent capabilities, which may be invalid. Looking at the code re-organisation I believe that having specific enum variants instead of bitfields will make the logic more explicit.

Reply:
As this is zenoh-pico, I see 8-bit as a target.
Also, you are just considering one capability, so you are not being fair in the analysis.

Today, we are considering 3 capabilities: `_Z_LINK_IS_STREAMED`, `_Z_LINK_IS_RELIABLE`, `_Z_LINK_IS_MULTICAST`.
Tomorrow we may have others capabilities like `_Z_LINK_IS_SECURE` or `_Z_LINK_IS_UNIDIRECTIONAL`.

I am still in favor of the bitfield, also because it scales better than to enumerate all the possible combinations.

Reply:
Let me know what is your decision, so I can remove or keep the requested changes in my previous code review.

Reply:
Just to weigh in on the discussion, the current bitfield structure isn't super well suited because only RELIABLE is a true flag/bool, for multicast and streamed it really represents enums:
```
enum LINK_CAPABILITY_TRANSPORT {
    UNICAST,
    MILTICAST,
    <New values>,
}
```
and
```
enum LINK_CAPABILITY_FLOW (just made this up) {
    DATAGRAM,
    STREAM,
}
```
Now there might be other flags in the future as you stated but in that case, if we really want to represent things correctly and use as little memory as possible then we have the possibility to do a register like structure:
```
struct capability {
    uint8_t transport: 2;
    uint8_t flow: 1;
    uint8_t reliable: 1;
    uin8_t reserved: 4;
}
```
It does open a lot of problems and future headache if we have to change the fields but that's a possibility.

Another one is to have a bitfield and a combined enum which is almost this PR but the bitfield is only the reliable flag.

Reply:
I do like the register-like structure proposal a bit better than the enum one. Not extremely familiar with it, but then couldn't we do
```c
struct capability {
    enum LINK_CAPABILITY_TRANSPORT transport: 2;
    enum LINK_CAPABILITY_FLOW flow: 1;
    _Bool reliable: 1;
    uin8_t reserved: 4;
}
```
to make each field unambiguous as to its purpose? (since C enums are dumb, this might backfire and make `capability` int-sized though)

Reply:
My main goal is to avoid ambiguity as well as avoiding the possibility to represent invalid states. For that reason, I believe the current PR is a good tradeoff, as mentioned by @jean-roland . 

I don't expect the enum to significantly grow and have a large number of variants. We are targeting fundamental characteristics of transports and links, which have an impact on the Zenoh protocol itself...